src/sdk.js

```diff
@@ -9,249 +9,257 @@
-*
-* @param {Object} options - clickHandler configuration object
-* @param {String} options.id - id of the element to add click handler to
-* @param {String} [options.state] - arbitrary parameter passed to redirect uri
-* @param {Boolean} [options.forcePrompt] - force permission approval screen to
-* show on every authentication, even if the user has previously consented
-* to the exact scope of permission
- */
-Smartcar.prototype.addClickHandler = function(options) {
-  const id = options.id;
-  const dialogOptions = {
-    state: options.state,
-    forcePrompt: options.forcePrompt,
-    development: this.development,
-  };
+    return link;
+  }
 
-  const element = document.getElementById(id);
-  if (!element) {
-    throw new Error(`Could not add click handler: element with id '${id}'` +
-      ' was not found.');
+  /**
+   * Launches the OAuth dialog flow.
+   *
+   * @param {Object} options - the link configuration object
+   * @param {String} [options.state] - arbitrary parameter passed to redirect uri
+   * @param {Boolean} [options.forcePrompt=false] - force permission approval
+   * screen to show on every authentication, even if the user has previously
+   * consented to the exact scope of permission
+   */
+  openDialog(options) {
+    const href = this.getAuthUrl(options);
+    const windowOptions = Smartcar._getWindowOptions();
+    window.open(href, 'Connect your car', windowOptions);
   }
 
-  element.addEventListener('click', () => {
-    this.openDialog(dialogOptions);
-    // this is equivalent to calling:
-    // event.preventDefault();
-    // event.stopPropogation();
-    return false;
-  });
-};
+  /**
+   * Adds an on-click event listener to the element with the provided id.
+   *
+   * On-click event calls openDialog when the specified element is clicked.
+   *
+   * @param {Object} options - clickHandler configuration object
+   * @param {String} options.id - id of the element to add click handler to
+   * @param {String} [options.state] - arbitrary parameter passed to redirect uri
+   * @param {Boolean} [options.forcePrompt] - force permission approval screen to
+   * show on every authentication, even if the user has previously consented
+   * to the exact scope of permission
+   */
+  addClickHandler(options) {
+    const id = options.id;
+    const dialogOptions = {
+      state: options.state,
+      forcePrompt: options.forcePrompt,
+      development: this.development,
+    };
+
+    const element = document.getElementById(id);
+    if (!element) {
+      throw new Error(
+        `Could not add click handler: element with id '${id}'` +
+          ' was not found.'
+      );
+    }
+
+    element.addEventListener('click', () => {
+      this.openDialog(dialogOptions);
+      // this is equivalent to calling:
+      // event.preventDefault();
+      // event.stopPropogation();
+      return false;
+    });
+  }
+}
 
 // expose AccessDenied error class
 Smartcar.AccessDenied = AccessDenied;
```

Comment:
we could also move the `AccessDenied` class def down here if we change the references to it as `smartcar.AccessDenied`